src/CattyTests/URLSessionMultipartExtensionTests.swift

```diff
@@ -0,0 +1,90 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+final class URLSessionMultipartExtensionTests: XCTestCase {
+
+    var urlSession: URLSession?
+    let httpBoundary = "---------------------------98598263596598246508247098291---------------------------"
+    let uploadParameterTag = "upload"
+    let fileChecksumParameterTag = "fileChecksum"
+    let tokenParameterTag = "token"
+    let projectNameTag = "projectTitle"
+    let projectDescriptionTag = "projectDescription"
+    let userEmailTag = "userEmail"
+    let userNameTag = "username"
+    let deviceLanguageTag = "deviceLanguage"
+    let statusCodeTag = "statusCode"
+    let projectIDTag = "projectId"
+    let availableTags = "constantTags"
+    var expectedZippedProjectData: Data!
+    var fileManagerMock: CBFileManagerMock!
+    var project: Project!
+
+    override func setUp() {
+        let header = Header.default()!
+        header.programName = kDefaultProjectBundleName
+        header.programDescription = ""
+        self.urlSession = URLSession.shared
+        self.project = Project()
```

Comment:
Please make this test a unit test. This test should not and does not have to depend on a project, or on the `UserDefaults` or on a `FileManager`. Please test your method by simply verifying the header and body of your task (e.g., `task.originalRequest.allHTTPHeaderFields` and `task.originalRequest.httpBody` and `String(decoding: task.originalRequest!.httpBody!, as: UTF8.self)`).